corehq/apps/domain/tests/test_views.py

```diff
@@ -111,6 +115,169 @@ def test_autocomplete_disabled(self):
+        cls.user = WebUser.create(cls.domain_name, cls.username, cls.password,
+                                  created_by=None, created_via=None, is_admin=True)
+
+        cls.domain_alert = cls._create_alert_for_domain(cls.domain_name, 'Test Alert 1!', cls.username)
+        cls.other_domain_alert = cls._create_alert_for_domain(cls.other_domain_name, 'Test Alert 2!', cls.username)
+
+        cls.url = reverse(ManageDomainAlertsView.urlname, kwargs={
+            'domain': cls.domain_name,
+        })
+
+    @staticmethod
+    def _create_alert_for_domain(domain, alert_text, username):
+        return CommCareHQAlert.objects.create(
+            text=alert_text,
+            domains=[domain],
+            created_by_domain=domain,
+            created_by_user=username
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain_alert.delete()
+        cls.other_domain_alert.delete()
+        cls.user.delete(cls.domain_name, deleted_by=None)
+        cls.domain.delete()
+        super().tearDownClass()
+
+    def setUp(self):
+        self.client = Client()
+        self.client.login(username=self.username, password=self.password)
+
+    def test_feature_flag_access_only(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 404)
+
+    @flag_enabled('CUSTOM_DOMAIN_BANNER_ALERTS')
+    def test_only_domain_alerts_listed(self):
+        alert = self.domain_alert
+
+        response = self.client.get(self.url)
+        self.assertListEqual(
+            response.context['alerts'],
+            [
+                {'active': False, 'html': 'Test Alert 1!', 'id': alert.id, 'created_by_user': self.username}
+            ]
+        )
+        self.assertEqual(response.status_code, 200)
+
+    @flag_enabled('CUSTOM_DOMAIN_BANNER_ALERTS')
+    def test_apply_command_with_missing_alert(self):
+        response = self.client.post(
+            self.url,
+            data={
+                'command': 'activate',
+                'alert_id': 0,
+            },
+        )
+        messages = list(get_messages(response.wsgi_request))
+        self.assertEqual(messages[0].message, 'Alert not found!')
+        self.assertEqual(response.status_code, 302)
+
+    @flag_enabled('CUSTOM_DOMAIN_BANNER_ALERTS')
+    def test_apply_command_with_invalid_command(self):
+        response = self.client.post(
+            self.url,
+            data={
+                'command': 'elevate',
+                'alert_id': self.domain_alert.id,
+            },
+        )
+        messages = list(get_messages(response.wsgi_request))
+        self.assertEqual(messages[0].message, 'Unexpected update received. Alert not updated!')
+        self.assertEqual(response.status_code, 302)
+
+    @flag_enabled('CUSTOM_DOMAIN_BANNER_ALERTS')
+    def test_apply_command_with_valid_command(self):
+        alert = self._create_alert_for_domain(self.domain, "New Alert!", self.username)
+        self.addCleanup(alert.delete)
```

Comment:
Cool!

https://github.com/dimagi/commcare-hq/pull/33592/commits/33d3e57763e11ec04685e8926f87bddf3972a19a


My understanding was that,
when we run tests locally where the tests don't stop on failure, this cleanup would be relevant.
I believe that if this test would fail the alert created will be retained and could conflict with another test.

Reply:
> I believe that if this test would fail the alert created will be retained and could conflict with another test.

Django's [`TestCase`](https://docs.djangoproject.com/en/4.2/topics/testing/tools/#testcase) will always roll back SQL database state, regardless of whether it passes or fails, running locally or on Github Actions (assuming the database backend is one like PostgreSQL that supports transactions).

Reply:
Ah, thank you.
I had it wrong.